### ISSUE TITLE
Support multiple link relationships

### ISSUE THREAD

**Issue by marcoscaceres** [WICG/webmonetization#192]
Web Monetization is currently limited to a single link relationship. However, we were wondering if we should just allow multiple link relationships and let the monetization agent figure out how to distribute payments amongst all the listed parties?

**Comment by marcoscaceres** [comment]
We could then get the "active" pointers from the Monetization interface via, for example, `.getPaymentPointers();`.

**Comment by AlexLakatos** [comment]
Talking to @sharafian about this, the complications involved would be significant, maybe we could leave this for a version 2 of the spec.

**Comment by stpeter** [comment]
It would indeed be interesting to find a solution here, though. For instance, a website might want to contribute its monetization "attention" to several different philanthropic organizations, not just one. But as you say, perhaps a version 2 feature...

**Comment by marcoscaceres** [comment]
I think it's ok to start with 1 and open it up once we figure out how.

**Comment by AlexLakatos** [comment]
> It would indeed be interesting to find a solution here, though. For instance, a website might want to contribute its monetization "attention" to several different philanthropic organizations, not just one.

That was the thought that spurred this conversation, and it looked like an easy decision. But after digging into it, the complexity of opening an ILP connection for each of them, and the potential for malicious behaviour (I could have a high number of links to the same pointer in the hope an unsuspecting web monetization provider will stream the same amount to all of them instead of splitting the amount between them) makes me think we're better off with one for now.

People have been using a random rotating pointer for monetizing the same content to multiple wallets, and while that's not a perfect solution (different users have different attention spans, and you'd need a mechanism more complex than random that rotates the payment pointer based on the attention each pointer has to ensure an even distribution) we could recommend people use it in the meantime.

**Comment by AlexLakatos** [comment]
On a related note, here's a current real-world example of how people monetize multiple payment pointers on the same page: https://www.youtube.com/watch?v=f0nCDVwOoHU

TL;DR: changing the content of the only link on the page (or as it is now, the meta tag)

**Comment by marcoscaceres** [comment]
The other way to do it is to create a bunch of iframes... that's also not really satisfactory.

**Comment by AlexLakatos** [comment]
The Coil extension at present splits the streaming rate equally between the iframes with payment pointers on a page. It seems to be a way for news outlets to share profits without doing all the heavy lifting of re-distributing funds.

**Comment by marcoscaceres** [comment]
The work around of creating many `<iframe>`s to get around the limitation of a single `<link>` is somewhat concerning: each `iframe` creates an entire browsing context, which consumes _a lot_ of resources.  

If the above indeed holds, then we should re-evaluate the single `<link>` limit per document, specially in light of hacking around the limitation with iframes. Naturally, there should be as many monetized iframes as needed for actual embedded content, but not being used to get around the single `<link>`-per-document limitation.

**Comment by sublimator** [comment]
Some related notes:

For `<iframe>`s (or demarcated content in general), I've often wondered if we could do something similar to what we are already doing with the visibility state at the "tab" level, and only stream to visible/active elements.  It would mean that the elements that garner the most attention get the pot. If you stuck to 1 stream per document you could choose based on ("most_visible", "element_offset") or the like. Would probably be easy to impl but a pain to push in a spec. 

We've never actually even prototyped something like that. I did have a branch where buttons were overlayed over iframes and borders drawn so people could choose the content they want to tip in the case of iframes. Much easier to just divvy the tip between the iframes though. Currently the tip just goes to the top frame ( which can be problematic when the top frame is not monetized )

**Comment by marcoscaceres** [comment]
That sounds like something that would be JS (e.g., using intersection observers) rather than something we would add to the spec. Then the content developers and host site can negotiate how/when they enable monetization.

**Comment by sublimator** [comment]
Right :)
